### ISSUE TITLE
CoC recommendations

### ISSUE THREAD

**Issue by drnikko** [drupaldiversity/administration#39]
This is a place for us to collect all of the language/changes/additions we'd like to see made to the Code of Conduct and that we discuss in #diversity-inclusion.

**Comment by Greg-Boggs** [comment]
Since no one else has responded yet, and it's being discussed on the Twitter, here's my go to source for CoC advice.

1. A clear license in addition to the site wide content license.
2. Descriptions of common but unacceptable behavior.
3. Information about enforcement.
4. Clear demarcation between anti-harassment policy and more general community guidelines.

There are now lots of open source projects we could model after. 

Source: http://geekfeminism.wikia.com/wiki/Code_of_conduct_evaluations

**Comment by catch56** [comment]
It's worth noting that page rates various CoCs on those four points, and we only get a tick for the guidelines on reporting (3), not (1), (2) or (4). There were a lot less examples and resources when we adopted it, and I know my opinion on some things has changed since the original discussions.

So a start would be - do we agree with the four points for rating? If so what changes would we need to make? Would it be easier to make tweaks to the existing CoC to include those things if so, or should we try to adopt a brand new one?

Also the Drupal Code of Conduct and the DrupalCon code of conduct are two different documents (something I'd not really registered).

**Comment by catch56** [comment]
Some things that for me are very obviously missing from the current code of conduct:

 - Drupal doesn't exist in a vacuum, there are existing forms of discrimination in society (gender, race, trans/non-binary gender identity, sexuality etc.) and we can name them. Can be 'including but not limited to'.

 - Some kind of indication that power is unevenly distributed in society, i.e. when talking about discrimination based on gender identity this means discrimination against trans and non-binary people, it doesn't mean discrimination against cis people because cis is the default identity and doesn't get discriminated against (and yes based on various conversations I think this needs to be spelled out...).

 - at least mentioning different categories of behaviour such as targeting groups, targeting specific individuals, feuds between individuals. The current CoC has always implied that it deals with disputes between individuals which of course happens but isn't the only thing that goes wrong. There might need to be different escalation/reporting guidelines for these too.

**Comment by beeradb** [comment]
I would recommend defining the scope of when people are being held to the CoC. Is it your everyday life, your public presence, your online presence, your interactions with the Drupal community? Currently, there doesn't seem to be a clear definition of what is "in scope" for the code of conduct.

**Comment by catch56** [comment]
Another thing that's not clear is what the boundaries of the Drupal community is.

This  issue tracker is not on Drupal.org, but it's all discussion about the Drupal community. Various Drupal modules are developed on github. All of the Drupal slack channels are as far as I know not official or run by the Drupal Association. There are various unofficial evening events at DrupalCon which are semi-open invitation to conference attendees. Some DrupalCamps get DA funding, some are completely independent. If you're discussing a Drupal topic on twitter or facebook with Drupal people is that the Drupal community? What about non-Drupal topics with Drupal people? What about non-Drupal topics with non-Drupal people?

Some of these questions are clear cut for me, some of them really aren't. If someone decides to call their local political representative a carpetbagger on twitter that's a bit different to calling a module maintainer a carpetbagger. But if they're involved in an ongoing public harassment campaign against a PHP developer from another project that's just as relevant as if the developer happened to be working on Drupal rather than Symfony or phpunit or something.

tl;dr I'm not sure it's possible to define where the Drupal community starts and ends, unless you limit it strictly to services and events the DA has strict control over which would exclude large sections of the community including spaces it's hard to avoid if you want to actively contribute to the project.

**Comment by ohthehugemanatee** [comment]
I think scope boundaries are not so important to define, actually. There is a separate CoC for drupal events - that one can and should be clear about where it applies (DA sponsored events, mostly). But the community-wide CoC has to be applied contextually. The goal is a set of shared values for how Drupalists treat each other when they're working together. It doesn't matter if we're working together on Github, Facebook, or in-person at Starbucks. 

When we're _not_ working together, everyone is free to be an asshole in their own way. I would say, including to other Drupalists. As much as I would love to censor all the assholes I know in the Drupal world, if one of our values is openness and inclusion, we can't do that. There has to be latitude for people to be assholes in ways that don't affect the project. And unfortunately, that's not a quantitative rule. It's a qualitative one.

In law, many countries handle those with an open system of courts to make the contextual judgment. In controversial cases, they even produce a document explaining their judgment. In our case, this is exactly the _raison d'être_ of the CWG. It is arguably not structured quite right, but that's a discussion for another thread.

Regarding naming specific limitations on the kinds of -isms we forbid; I think that strays into dangerously culturally-centric territory. For example, when you say that cis-gendered people can't face discrimination because they are a numerical majority group, that's not always the case. A lot of it depends on your limitations. For example, in some communities of my city cisgenders are definitely the minority, and at some events they're an extreme minority. Where's the demarcation? If the only cis-gender at a predominantly LGBT event feels attacked and shamed, isn't that something we should care about? What about discrimination against a caucasian person at an event in Mumbai? 

Ultimately, we want to forbid ANY discriminatory behavior, even things that would be considered reverse discrimination in our home cultures.

**Comment by stephenpurkiss** [comment]
FYI here's the [Drutopia Code of Conduct](https://gitlab.com/drutopia/drutopia/wikis/drutopia-code-of-conduct) which says it's largely taken from [The Citizen Code of Conduct distributed by Stumptown Syndicate](http://stumptownsyndicate.org/code-of-conduct/) which in turn says portions of text derived from the [Django Code of Conduct](https://www.djangoproject.com/conduct/) and the [Geek Feminism Anti-Harassment Policy](http://geekfeminism.wikia.com/wiki/Conference_anti-harassment/Policy).

**Comment by catch56** [comment]
"For example, when you say that cis-gendered people can't face discrimination because they are a numerical majority group, that's not always the case."

I specifically didn't say 'numerical majority group', I said 'default identity'. Once you switch from what is default or has power at a societal level to pure numbers in any random social situation you end up with these kinds of false equivalences. Subjugation of women has not been based on men being a numerical majority, apartheid South Africa was based on white minority rule; it's not about numbers as such.

If you go to an event in your town and happen to be the only cis-gendered person, it's probably not going to make any difference to your general access to housing, employment, healthcare, education etc., because you still live in a society where cis the default and where trans and non-binary people have uneven access to those.If someone was rude to you or even physically attacked you, that's not nice, but it's not systemic discrimination. This is why I mentioned different categories of behaviour rather than lumping everything in together. Things can be bad but also different.

"What about discrimination against a caucasian person at an event in Mumbai?"
Caste in India is an example of social stratification which isn't on most American people's default list of -isms, there's also discrimination against Muslims (which is on the increase with the election of Modi). There are real examples that are culturally specific without needing to bring in 'reverse racism against white people'. Examples can be useful but hypotheticals often are not. If there's a case that's not covered, 'including but not limited to' and allowing for some discretion in the CWG's handling helps rather than refusing to give any examples at all. I argued against examples in the discussion of the original CoC draft ('what if you miss one?' etc.) and no longer agree with that position.

@stevepurkiss thanks for the link to Drutopia's CoC, that generally seems to have a good balance. I noticed Geek Feminism has separate community and conference policies now - community policy is here: http://geekfeminism.wikia.com/wiki/Community_anti-harassment/Policy

**Comment by stephenpurkiss** [comment]
Hi - I'm wondering how CoC deals with, if at all, the conduct of companies both in terms of community in general and events. From everything I've seen following these discussions so far and the surrounding recent events the focus is on individuals as opposed to people acting in a group.

One of the main reasons I ask is because we still have radio silence on continued requests for response to the contents of this blog (use LinkedIn if you need to know which is A and which is B):

https://nateofnine.com/2016/12/04/a-tale-of-two-communities/

When I previously complained about this the response I received was "it's not our remit". To me, that answer is not good enough, it's people who run these companies, people within the community, their actions reflect our community on the outside yet according to the current setup there is one rule for "us" and another for "them".

Here's my previous convo with CWG:

Adam Hill <adam@consultanddesign.com>
24 Jan

to me, drupal-cwg 
Dear Steve,

Firstly a huge apology for not getting back to you sooner on this - with the festive season and a few too many responsibilities recently I wasn't able to get back to you. I hope this finds you well.

Secondly thank you for reaching out to the Community Working Group on this issue. We know how much it means to you and others that the Drupal Community is welcoming and inclusive. The contents of Nate's blog post are of course worrying and did not go unnoticed in our discussions over the last few weeks.

The core role of the Community Working Group is to help mediate conflicts or breaches of the Code of Conduct by community members. With this remit, there is therefore not so much that we can do directly in response. Nate is aware of our group but has not chosen to report any of these activities to us directly and as he details in his comments, "It truly was no single event."

We also have no control over the actions of companies that support or align themselves with Drupal.org or the Drupal Association, nor the power to do anything about their behaviour as it stands. Though rare, we have had specific reports of actions carried out by specific companies and have acted accordingly where it is not an internal company matter. Such situations have resulted in public apologies and conflict resolution.

If you have specific examples where individuals or companies have broken the code of conduct we will, through our conflict resolution process, take action on these matters. We have no problem if you help to encourage others to report issues they have had. We cannot however comment on the issues Nate details due to no request to get involved, and more importantly no clear information about those involved.

It isn't our intention to make judgements or take stands against people or companies, nor to do any kind of PR work. We aim only to carry out our processes as best we can and where possible carry out activities with the aim of more generally improving the health of the community.

We are deeply concerned by the general feeling from this post about people feeling uncomfortable and alienated by the actions of individuals and companies within the community. Using our limited resources outside of dealing with individual cases we are looking at ways to improve this, but we are open to ideas that could help to prevent this kind of problem.

If you would like to contribute to that process we would be more than happy to have some feedback  and input about things we could do to make the community more welcoming within our remit and I'd love the opportunity to talk to you about this further in person over Hangout. I'll also be at DrupalCamp London if you'd like to spend some time there talking over ideas, assuming you’ll be attending.

While I realise it is not exactly what you would have liked to hear, I hope the above provides some insight into our position and approach. Do let me know if you'd like to discuss this further.

All the best,

Adam

----------

Adam Hill - Operations Director
Consult and Design International Limited

+44 (0) 191 645 1111
http://www.consultanddesign.com

Join a new meetup for the North East digital community, Sunderland Digital
Consult and Design have setup the first Drupal Hub in Sunderland

Sunderland Office: Software Centre, Tavistock Place, Sunderland, SR1 1PB, UK
Latvia Office: Oskara Kalpaka iela 107-16, Liepāja, LV-3405, Latvia
Registered Office: International House, 32 Edwards Road, Tyne and Wear, NE26 2BJ

Registered Company in England & Wales No. 05491181

On 8 December 2016 at 10:51, Steve Purkiss <steve@purkiss.com> wrote:
Hi Team CWG,

I, and I'm sure many of you too, read a disturbing blog post recently:

https://nateofnine.com/2016/12/04/a-tale-of-two-communities/

In this post the activities of two communities named A and B are
described. The author removed himself from community A due to the
activities of a few, small-minded people who seem to have brought
community A into much disrepute. That community A is Drupal, as I'm
sure you are aware.

According to the author:

"Being told upon starting a job at one of the prominent companies that
I should be prepared to take my clients to strip clubs, and later
watching it actually happen as part of conference culture, even part
of sales expense accounts."

 and

"Arguments between companies that can only be described as
religious-style battles (in public and at conferences) based over who
was right without asking end users what they desired or experienced."

and

"I personally observed a company prominent in both communities chose
to react to a (potentially legitimate) disagreement by attempting to
rally anger and protest against the facilitators in the midst of an
event in Community B and then cried foul when they were banned from
the event for inappropriate behavior. I then saw numerous members of
Community A actively promoting and justifying the company’s
misbehavior by saying someone else acted poorly as well."

of which that last I saw occur in realtime over social media.

There are many other allegations in the post. A simple trip to
LinkedIn will of course reveal the first company discussed, a simple
search on social media reveals the second. I say it like this because
of course I and you know who it is, but it is also incredibly easy for
anyone else not so involved.

It is no wonder many prefer not to get involved in the community when
this is the sort of activities that go on from the roots. We should
not be supporting companies who have this kind of behaviour, we should
not be promoting them through the *.drupal.org infrastructure, we
should not be encouraging people to work for them and continue this
sort of behaviour, and we certainly should not be creating methods and
practices of governing our community which favour this behaviour.

If the CWG's intention is to include, and support the Drupal community
as a whole then instead of responding to emails like this with another
PR job, a stand needs to be made. A stand that says we do not accept
this behaviour, it has gone on for too many years and we are now going
to change this because no matter what immediate problems it causes for
some, those some ruin it for the rest of us - the larger silent
majority who are trying to push ahead in terms of technology,
co-operation, inclusion and the furthering of humanity.

Don't miss this opportunity to make a stand and make a change, please.

Thank you for your time, I will keep my fingers crossed.

Steve
--
You received this message because you are subscribed to the Google Groups "Drupal Community Working Group" group.
To post to this group, send email to drupal-cwg@googlegroups.com.
Visit this group at https://groups.google.com/group/drupal-cwg.

**Comment by stephenpurkiss** [comment]
Just in case you think otherwise, I posted this not to stir things up but because I truly believe this is an issue we need to address when talking about diversity and inclusion, and at the moment I don't see that being done.

**Comment by stephenpurkiss** [comment]
I'd also like to pick up on a point made over on another issue which at the top says it shouldn't be used for discussing but for listing other texts to use so dropping it in here.

https://github.com/drupaldiversity/administration/issues/41#issuecomment-292499768

The wording is 'professional' and whilst many including myself use Drupal in my profession, I notice a couple of people posting recently who said they're helping out their local group or church or charity etc. and it's not necessarily their profession so I'd be hesitant to use the term 'professional community' if indeed we are to cover the whole of the Drupal community.

We're all people.

**Comment by jdegoes** [comment]
@stevepurkiss The use of the word professional doesn't necessarily mean you earn a living from it (that is a discredit to all the volunteers who work in our industry — do they not have the capability to be professional? I'd say absolutely, it doesn't matter they're not getting paid for it!). It is simply to contrast with personal, to indicate that goals are of a professional nature, and that behavior is expected to be professional. In personal communities, we often show favoritism, inviting only people we like; we sometimes impose ideological requirements (for example, only allow Christians to attend a prayer study); and we often favor homogeneity, rather than encouraging diversity.

**Comment by drnikko** [comment]
We added a contributor code of conduct to the repo and will be closing this for now.  Creating a community-wide code of conduct is 100% out of our scope and out of our domain of control.  IIRC, this is something that is being done in the d.o queue for governance and/or the cwg.
